spec.md

```diff
@@ -1,81 +1,401 @@
+journalistic practices of quotations and other such forms.
+
+However, due to how the signature is generated, no overlay can be present on
+these quotes and they are limited to specific resolutions which have been
+signed by the original party. This then also can cause issues for streaming
+platforms, who usually compress videos to reduce storage cost and also change
+the resolution depending on the users internet connection or preference.
+Therefore, it is then up to the platforms to validate that the given content is
+already compressed to their satisfaction and signed for all potential
+resolutions for streaming.
+
+This also brings up a further issue, which is that if you are signing for
+multiple resolutions, the compression algorithm you used must be known to the
+streaming platform and deterministic -- although most are.
+
+#### Signing information
+
+As we are signing in chunks connected to timestamps, the signatures themselves
+will be stored inside a subtitles file, one for each resolution it is being
+signed at.
+
+Each timeframe of the subtitles file will contain a JSON-LD object which describes
+the id of the credentials which generated the signature as well as the pixel
+area which has been signed. This object should look as such:
+
+<!-- TODO: This may need to be updated to another more efficient dataformat -->
+
+```json
+[
+  // An array of Signature objects
+  {
+    "pos": {
+      // Vector object determining where the position starts
+      "x": 0, // Unsigned integeter between 0 and the width of the video
+      "y": 0 // Unsigned integeter between 0 and the height of the video
+    },
+    "size": {
+      // Vector object determining the size of the area which has been signed
+      "x": 1920, // Unsigned integeter between 0 and the width - pos.x of the video
+      "y": 1080 // Unsigned integeter between 0 and the height - pos.y of the video
+    },
+    "credential": {
+      // If the credential already has been defined, we can simple refer to its ID number
+      // defined in its definition
+
+      "id": "http://example.edu/credentials/1872", // The ID specified when defining a credential
+    } | {
+      "@context": [
+        "https://www.w3.org/2018/credentials/v1",
+        "https://www.w3.org/2018/credentials/examples/v1"
+      ],
+      "id": "http://example.edu/credentials/1872",
+      "type": ["VerifiableCredential", "AlumniCredential"],
+      "issuer": "https://example.edu/issuers/565049",
+      "issuanceDate": "2010-01-01T19:23:24Z",
+      "credentialSubject": {
+        "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+        "alumniOf": {
+          "id": "did:example:c276e12ec21ebfeb1f712ebc6f1",
+          "name": [{
+            "value": "Example University",
+            "lang": "en"
+          }, {
+            "value": "Exemple d'Université",
+            "lang": "fr"
+          }]
+        }
+      },
+      "proof": {
+        "type": "RsaSignature2018",
```

Comment:
I will be using dilithium in the libraries and so it should be represented here.